cmake/modules/vif.cmake

```diff
@@ -0,0 +1,36 @@
+# Copyright (c) 2022 The Chromium OS Authors
+# SPDX-License-Identifier: Apache-2.0
+
+# Generates VIF policies in XML format from zephyr.dts file generated during build.
+
+set(GEN_VIF_SCRIPT ${ZEPHYR_BASE}/scripts/utils/generate_vif/generate_vif.py)
+set(ZEPHYR_DTS_PRE ${PROJECT_BINARY_DIR}/zephyr.dts.pre)
+set(DTS_BINDINGS ${ZEPHYR_BASE}/dts/bindings)
+set(DTS_COMPATIBLE usb-c-connector)
+set(VIF_XML ${PROJECT_BINARY_DIR}/vif.xml)
+
+message(STATUS "executing gen vif, reading file from loc : ${ZEPHYR_DTS}")
+
+# Executes generate_vif.py to create vif.xml.
+
+set(CMD_GEN_VIF ${PYTHON_EXECUTABLE} ${GEN_VIF_SCRIPT}
+        --dts ${ZEPHYR_DTS_PRE}
+        --vif-out ${VIF_XML}.new
+        --bindings-dirs ${DTS_BINDINGS}
+        --compatible ${DTS_COMPATIBLE}
+        )
+
+execute_process(
```

Comment:
Does this work? I thought that `execute_process` will happen at config time for CMake. I think you need to create a target:

```
add_custom_target(
  vif_target
  COMMAND ${CMD_GEN_VIF}
  DEPENDS ${ZEPHYR_DTS_PRE}
  WORKING_DIRECTORY ${PROJECT_BINARY_DIR}
  RESULT_VARIABLE reg
)
```

You might need to tweak the `DEPENDS` line there.

Reply:
I actually followed the way gen_defines.py generates zephyr.dts from zephyr.dts.pre at the same folder vif.xml is being generated : https://github.com/zephyrproject-rtos/zephyr/blob/main/cmake/modules/dts.cmake#L210